src/Features/CSharp/Portable/RemoveUnusedParametersAndValues/CSharpRemoveUnusedValuesCodeFixProvider.cs

```diff
@@ -0,0 +1,235 @@
+
+                        // Process the discard variable declaration to replace it
+                        // with an assignment to discard.
+                        GenerateAssignmentForDiscardVariable(variable);
+                    }
+                }
+
+                // Process all the remaining variable declarators to generate
+                // a local declaration statement with these variables.
+                GenerateDeclarationStatementForCurrentNonDiscardVariables();
+
+                // Now replace the original local declaration statement with
+                // the replacement statement list tracked in _statementsBuilder.
+
+                if (_statementsBuilder.Count == 0)
+                {
+                    // Nothing to replace.
+                    return;
+                }
+
+                // Move the leading trivia from original local declaration statement
+                // to the first statement of the replacement statement list.
+                var leadingTrivia = _localDeclarationStatement.Declaration.Type.GetLeadingTrivia()
+                    .Concat(_localDeclarationStatement.Declaration.Type.GetTrailingTrivia());
+                _statementsBuilder[0] = _statementsBuilder[0].WithLeadingTrivia(leadingTrivia);
+
+                // Move the trailing trivia from original local declaration statement
+                // to the last statement of the replacement statement list.
+                var last = _statementsBuilder.Count - 1;
+                var trailingTrivia = _localDeclarationStatement.SemicolonToken.GetAllTrivia();
+                _statementsBuilder[last] = _statementsBuilder[last].WithTrailingTrivia(trailingTrivia);
+
+                // Replace the original local declaration statement with new statement list
+                // from _statementsBuilder.
+                if (_localDeclarationStatement.Parent is BlockSyntax)
+                {
+                    _editor.InsertAfter(_localDeclarationStatement, _statementsBuilder.Skip(1));
+                    _editor.ReplaceNode(_localDeclarationStatement, _statementsBuilder[0]);
+                }
+                else
+                {
+                    _editor.ReplaceNode(_localDeclarationStatement, SyntaxFactory.Block(_statementsBuilder));
+                }
+            }
+
+            private void GenerateDeclarationStatementForCurrentNonDiscardVariables()
+            {
+                // Convert a discard declaration with initializer of the form "var _ = M();" into
+                // a discard assignment "_ = M();"
+                if (_currentNonDiscardVariables.Count > 0)
+                {
+                    var statement = SyntaxFactory.LocalDeclarationStatement(
+                                        SyntaxFactory.VariableDeclaration(_localDeclarationStatement.Declaration.Type, _currentNonDiscardVariables))
+                                    .WithAdditionalAnnotations(Formatter.Annotation);
```

Comment:
having trouble reconciling this code with the comment.  You say it's making a discard assignment, but i can see a variable-declaration being made here witha  type.

Reply:
Oops, copied comment at the wrong location. Fixed now, thanks!

---
In reply to: [234799909](https://github.com/dotnet/roslyn/pull/30777#discussion_r234799909) [](ancestors = 234799909)